### PR TITLE
Checkbox: Using the max attribute can limit the number of items that …

### PR DESCRIPTION
**Pull request by masongzhi**
…can be checked (#14742)

Please make sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md) | [Español](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

Checkbox: Using the max attribute can limit the number of items that can be checked (#14742)

### CODE REVIEW

**Comment by luckyCao** [comment]
@masongzhi 
https://deploy-preview-15585--element.netlify.com/#/zh-CN/component/checkbox#ke-xuan-xiang-mu-shu-liang-de-xian-zhi
有 bug
实际：反选北京，上海变成 disabled
期望：反选北京，其他选项都应该是可选状态

**Comment by masongzhi** [comment]
@luckyCao `min=1、max=2`，我认为是至少选一个，所以剩下一个选项时不能取消选择

File: packages/checkbox/src/checkbox-button.vue [hunk 1]
```diff
@@ -152,7 +152,7 @@
 
       isDisabled() {
         return this._checkboxGroup
-          ? this._checkboxGroup.disabled || this.disabled || (this.elForm || {}).disabled
+          ? this._checkboxGroup.disabled || this.disabled || (this.elForm || {}).disabled || (this.model.length >= this._checkboxGroup.max && !this.isChecked) || (this.model.length <= this._checkboxGroup.min && this.isChecked)
```
Comment (luckyCao) [inline packages/checkbox/src/checkbox-button.vue]: 这个判断全部放在一起，比较难以维护，优化一下吧

**Comment by masongzhi** [comment]
@luckyCao 抽出了`isLimitDisabled` computed

**Pull request merged**